packages/engine-server/src/backup/service.ts

```diff
@@ -68,8 +69,8 @@ export class BackupService implements Disposable, IBackupService {
       root: this.wsRoot,
     });
     await fs.ensureDir(this.backupRoot);
-    Object.keys(BackupKeyEnum).forEach(async (key) => {
-      await fs.ensureDir(path.join(this.backupRoot, key));
+    await asyncLoopOneAtATime<string>(Object.values(BackupKeyEnum), (key) => {
+      return fs.ensureDir(path.join(this.backupRoot, key));
```

Comment:
In this case, do we actually need the logic in the loop to be sequential though?

i.e. versus
```typescript
    await Promise.all(
      Object.keys(BackupKeyEnum).map((key) =>
        fs.ensureDir(path.join(this.backupRoot, key))
      )
    );
```

Reply:
that's fair. updated